homeassistant/components/zwave_js/binary_sensor.py

```diff
@@ -356,3 +380,43 @@ def _get_sensor_mapping(self) -> NotificationSensorMapping:
                 mapping_info = mapping.copy()
                 return mapping_info
         return {}
+
+
+class ZWavePropertyBinarySensor(ZWaveBaseEntity, BinarySensorEntity):
+    """Representation of a Z-Wave binary_sensor from a property."""
+
+    def __init__(
+        self, config_entry: ConfigEntry, client: ZwaveClient, info: ZwaveDiscoveryInfo
+    ) -> None:
+        """Initialize a ZWavePropertyBinarySensor entity."""
+        super().__init__(config_entry, client, info)
+        # check if we have a custom mapping for this value
+        self._mapping_info = self._get_sensor_mapping()
+
+    @property
+    def is_on(self) -> bool:
+        """Return if the sensor is on or off."""
+        return self.info.primary_value.value in self._mapping_info["on_states"]
+
+    @property
+    def device_class(self) -> Optional[str]:
+        """Return device class."""
+        return self._mapping_info.get("device_class")
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        # We hide some more advanced sensors by default to not overwhelm users
+        # unless explicitly stated in a mapping, assume deisabled by default
+        return self._mapping_info.get("enabled", False)
+
+    @callback
+    def _get_sensor_mapping(self) -> PropertySensorMapping:
+        """Try to get a device specific mapping for this sensor."""
+        mapping_info = PropertySensorMapping()
```

Comment:
What does this result in? An empty dict?

Reply:
yeah. it was the only way i could get all the linting tests to pass and get code coverage and not invent some bogus ZWaveDiscoverySchema without a corresponding PropertySensorMapping.

Reply:
Ok.